modules/Auth/Services/AuthService.php

```
@@ -161,13 +161,22 @@ private function sendOtpByStep($step, $identifier)
 
     private function checkOtpByStep($step, $identifier, $otp)
     {
-        if ($step->login_option == "otp" && (new Otp)->validate($identifier, $otp)->status == false) {
-            throw new \Exception(__("validation.invalid-otp"), 401);
+        if (empty($step)) {
+            return true;
+        }
+        if ($step->login_option == "otp") {
+            return (new Otp)->validate($identifier, $otp)->status;
```

The OTP validation should still throw an exception when invalid, similar to the previous implementation. Currently, it silently returns `false` which might lead to unexpected behavior in the calling code.